app/scripts/lib/app-start.js

```diff
@@ -543,29 +544,26 @@ function (
     },
 
     allResourcesReady: function () {
-      var self = this;
-      return this._selectStartPage()
-          .then(function (startPage) {
-            // The IFrame cannot use pushState or else a page transition
-            // would cause the parent frame to redirect.
-            var usePushState = ! self._isInAnIframe();
-
-            if (! usePushState) {
-              // If pushState cannot be used, Backbone falls back to using
-              // the hashchange. Put the initial pathname onto the hash
-              // so the correct page loads.
-              self._window.location.hash = self._window.location.pathname;
-            }
-
-            // If a new start page is specified, do not attempt to render
-            // the route displayed in the URL because the user is
-            // immediately redirected
-            var isSilent = !! startPage;
-            self._history.start({ pushState: usePushState, silent: isSilent });
-            if (startPage) {
-              self._router.navigate(startPage);
-            }
-          });
+      // The IFrame cannot use pushState or else a page transition
+      // would cause the parent frame to redirect.
+      var usePushState = ! this._isInAnIframe();
+
+      if (! usePushState) {
+        // If pushState cannot be used, Backbone falls back to using
+        // the hashchange. Put the initial pathname onto the hash
+        // so the correct page loads.
+        this._window.location.hash = this._window.location.pathname;
```

Comment:
Not for this PR, but history and routing seem intimately related. Setting up history seems like a task for the router to me.